### ISSUE TITLE
Random user-agent

### ISSUE THREAD

**Issue by arall** [Arachni/arachni#621]
I think that it would be a good to have an option to use a random 'real' user agent, like SQLMap does, in order to bypass some WAFs easily, instead of passing super long parameters like:

```
--http-user-agent "Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.117 Safari/537.36"
```

**Comment by Zapotek** [comment]
There are a few issues with that approach:

1. Can lead to decreased coverage.
   1. Invalidated session when same cookies are provided by multiple user-agents.
   2. Account lockout features.
   3. Different responses sent depending on UA -- mobile, tablet, IE fixes, etc.
2. The scan becomes non-reproducible -- see 1.
3. You're actually testing the WAF rather than the webapp, that's out of the project's scope.
4. You can't rely on those results for security.
  1. WAF protects you for a while.
  2. Bug is found in WAF.
  3. All the results you had 'till then become meaningless.
  4. False sense of security is gone and you're completely exposed.

In short, Arachni is a **web application** security scanner so it's better if it's used for the task it was designed.

Using multiple user agents is good for coverage but only when you perform a full scan with each one, then you can contrast and compare and gain a lot of useful information out of it.

I don't think this is a good idea.
